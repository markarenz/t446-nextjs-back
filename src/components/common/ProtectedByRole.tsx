import { useSession } from 'next-auth/react';
import { getIsAdmin } from '../../helpers';

type Props = {
  children: JSX.Element;
};
const ProtectedByRole: React.FC<Props> = ({ children }) => {
  const { data: session } = useSession();
  const isAuthorized = session ? getIsAdmin(session) : false;
  return isAuthorized ? (
    children
  ) : (
    <div>
      <h2>You are not authorized for this page.</h2>
    </div>
  );
};

export default ProtectedByRole;
