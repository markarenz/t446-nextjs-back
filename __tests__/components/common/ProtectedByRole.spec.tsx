import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import ProtectedByRole from '../../../src/components/common/ProtectedByRole';

jest.mock('next-auth/react', () => ({
  useSession: jest
    .fn()
    .mockReturnValueOnce({
      data: {
        user: {
          role: ''
        }
      }
    })
    .mockReturnValueOnce({})
    .mockReturnValue({
      data: {
        user: {
          role: 'admin'
        }
      }
    })
}));

describe('ProtectedByRole', () => {
  it('renders warning when unauthorized', async () => {
    render(
      <ProtectedByRole>
        <div>OK</div>
      </ProtectedByRole>
    );
    const element = screen.queryByTestId('unauthorized');
    expect(element).toBeInTheDocument();
  });
  it('renders warning when session is null', async () => {
    render(
      <ProtectedByRole>
        <div>OK</div>
      </ProtectedByRole>
    );
    const element = screen.queryByTestId('unauthorized');
    expect(element).toBeInTheDocument();
  });

  it('renders component when authorized', async () => {
    render(
      <ProtectedByRole>
        <div>OK</div>
      </ProtectedByRole>
    );
    const element = screen.queryByTestId('unauthorized');
    expect(element).toBe(null);
  });
});
