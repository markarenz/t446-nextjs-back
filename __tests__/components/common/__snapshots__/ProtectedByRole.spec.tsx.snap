// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedByRole renders warning when unauthorized 1`] = `
<div
  data-testid="unauthorized"
>
  <h2>
    You are not authorized for this page.
  </h2>
</div>
`;
